### PR TITLE
chore: adds engines field on package.json

### PR DESCRIPTION
**Pull request by helio-frota**

### CODE REVIEW

**Review by richardlau — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -20,6 +20,9 @@
   "bin": {
     "npcheck": "index.js"
   },
+  "engines": {
+    "node": "^16 || ^14"
```
Comment (mhdawson) [inline package.json:24]: Is there something that say anything higher than 14? I don't think we want to block running on newer versions?

Comment (helio-frota) [reply package.json:24]: npcheck is using some JS features that are not available on node 12.x, for example `?.`  optional chaining.

Comment (richardlau) [reply package.json:24]: I think we could say 
```suggestion
    "node": ">=14"
```
FWIW `npm` doesn't block by default (I think it does warn). `yarn` (or at least some versions of `yarn`) did block unless a command line flag was passed.

Comment (helio-frota) [reply package.json:24]: `"node": ">=14"` looks better than (each) explicit version

**Review by richardlau — APPROVED** [review]

**Pull request merged**